scipy/interpolate/_bsplines.py

```diff
@@ -297,6 +298,75 @@ def basis_element(cls, t, extrapolate=True):
         c = np.zeros_like(t)
         c[k] = 1.
         return cls.construct_fast(t, c, k, extrapolate)
+    
+    @classmethod
+    def design_matrix(cls, x, t, k, kind="CSR"):
```

Comment:
I would have expected either a method such that I can call `BSpline(...).design_matrix(x)` as a BSpline object already contains `t` and `k` or a standalone function, but not a classmethod.
The fact that `cls` is not used might underline my point.

Reply:
Talking about the standalone function: creating `design_maitrx` function as a class method underlines the fact that a design matrix is built for BSpline interpolant which makes sense if we have `design_matrix` functions in different parts of SciPy (in interpolation in power basis for instance). 
Making a method such as `BSpline(...).design_matrix(x)` would require the user to know the vector of unknowns `c` to create a BSpline object to make a design matrix which doesn’t even depend on this vector.
For these reasons I'm not sure if structure you mentioned are needed.